core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -172,7 +172,7 @@ private class ShuffleStatus(numPartitions: Int) extends Logging {
    */
   def removeOutputsOnExecutor(execId: String): Unit = withWriteLock {
     logDebug(s"Removing outputs for execId ${execId}")
-    removeOutputsByFilter(x => x.asInstanceOf[BlockManagerId].executorId == execId)
+    removeOutputsByFilter(x => x.asInstanceOf[ExecutorLocation].executorId == execId)
```

Comment:
Could we do `x.isInstanceOf[ExecutorLocation] && x.asInstanceOf[ExecutorLocation].executorId == execId`, since people could have `Location` implementation which does not implement `ExecutorLocation`?

Reply:
Same to https://github.com/apache/spark/pull/31876#discussion_r618052332.